packages/wallet/src/redux/__tests__/helpers.ts

```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
     expect(item.messagePayload.type).toEqual(messageType);
   });
 };
-export const expectThisMessageAndCommitmentSent = (
-  state: SideEffectState,
-  c: Partial<Commitment>,
-  messageType: string,
-) => {
-  expectSideEffect('messageOutbox', state, item => {
-    expect(item.messagePayload.type).toEqual(messageType);
-    expect(item.messagePayload.signedCommitment.commitment).toMatchObject(c);
-  });
-};
 
-export const expectThisMessageAndTheseCommitmentsSent = (
+type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
+
+function transformCommitmentToMatcher(sc: { commitment: Partial<Commitment>; signature?: string }) {
+  if (sc.signature) {
+    return expect.objectContaining({ signature: sc.signature });
+  } else {
+    return expect.objectContaining({ commitment: expect.objectContaining(sc.commitment) });
+  }
+}
+
+export const expectThisCommitmentSent = (
   state: SideEffectState,
-  commitments: PartialCommitments,
-  messageType: string,
+  commitment: Partial<Commitment>,
+  type = 'WALLET.COMMON.COMMITMENT_RECEIVED',
+  idx = 0,
 ) => {
-  expectSideEffect('messageOutbox', state, item => {
-    expect(item.messagePayload.type).toEqual(messageType);
-    expect(item.messagePayload.signedCommitments).toMatchObject(
-      // If the user passes a signature, we should match against it. Otherwise,
-      // the signature should be present, but we don't care what its value is
-      commitments.map(({ commitment, signature }) => ({
-        commitment,
-        signature: signature || expect.any(String),
-      })),
+  const messageOutbox = getOutboxState(state, 'messageOutbox');
+
+  try {
+    // Passes when at least one message matches
+    // In the case of multiple messages queued, this approach does not care about
+    // their order, which is beneficial.
+    // However, the diffs produced by jest are unreadable, so when this expectation fails,
+    // we catch the error below
+    expect(messageOutbox).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          messagePayload: expect.objectContaining({
+            signedCommitment: transformCommitmentToMatcher({ commitment }),
+          }),
+        }),
+      ]),
     );
-  });
-};
+  } catch (err) {
+    if ('matcherResult' in err) {
+      // In this case, we've caught a jest expectation error.
+      // We try to help the developer by using expect(foo).toMatchObject(bar)
+      // The errors are much more useful in this case, but will be deceiving in the case when
+      // multiple messages are queued.
+
+      // To help with debugging, you can change the idx variable when running tests to 'search'
+      // for the correct commitment
 
-export const expectThisCommitmentSent = (state: SideEffectState, c: Partial<Commitment>) => {
-  expectThisMessageAndCommitmentSent(state, c, 'WALLET.COMMON.COMMITMENT_RECEIVED');
+      expect(messageOutbox[idx]).toMatchObject({
```

Comment:
What about a `console.warn` message that points the developer to this code. Something like: 'Make sure to understand the matcher before diving into debugging'?